### PR TITLE
[batch] pass credentials for both GCR and Artifact Registry images

### PR DESCRIPTION
**Pull request by lgruen**
The Artifact Registry is useful to avoid network egress costs for regions like Australia, where GCR doesn't provide local hosting.

Partially addresses #9872.

### CODE REVIEW

File: batch/batch/worker/worker.py [hunk 1]
```diff
@@ -341,7 +341,7 @@ async def get_container_status(self):
     async def run(self, worker):
         try:
             async with self.step('pulling'):
-                if self.image.startswith('gcr.io/'):
+                if self.image.startswith('gcr.io/') or 'docker.pkg.dev/' in self.image:
```
Comment (danking) [inline batch/batch/worker/worker.py:344]: hmm, we really want this to be the host portion of the image not the whole image.

Can we do instead:
```
image_parts = image.split('/')
image_hostname = ''
if len(image_parts) > 1:
    image_hostname = image_parts[0]
if image_hostname == 'gcr.io' or image_hostname.endswith('docker.pkg.dev'):
```

**Review by danking — CHANGES_REQUESTED** [review]

File: batch/batch/worker/worker.py (hunk 1 above)
Comment (lgruen) [reply batch/batch/worker/worker.py]: Good point! Moved this to a small utility function.

**Review by danking — APPROVED** [review]

**Comment by danking** [comment]
Thanks for the improvement!

**Comment by lgruen** [comment]
Thanks for the review! Could you please also merge the PR, as I don't have the necessary permissions?

**Comment by danking** [comment]
Bah, sorry, CI does all merges and there's a bug introduced in a python dependency of ours. I'll get that fixed.

**Comment by danking** [comment]
Ah, the fix was just merged, CI should test and merge this within 30 minutes.

**Comment by lgruen** [comment]
Awesome, thanks a lot for debugging!

**Pull request merged**